### DISCUSSION TITLE
How to implement branching activities in elsa 3.0  when workflow is developed in programmatic way by implementing IWorkflow

### DISCUSSION THREAD

**Discussion by NiveditaGalwade** [elsa-workflows/elsa-core#5589] [Q&A]
Workflow is implemeted in below way
public class CustomWorkflow : IWorkflow
{
    public void Build(IWorkflowBuilder builder)
    {
        builder
             .StartWith<Activity1>()
             .Then<Activity2>()
             
             .Then<Activity3>()
             .Then<Activity4>();
    }
}

I wanted to apply condition before executing Activity3. how to apply some flag and execute it based on flag value. If flag true execute Activity3

**Comment by glime-ai** [comment]
To apply a condition before executing `Activity3`, you can use the `If` or `Switch` activity provided by Elsa to branch your workflow based on a flag value. Here's an example of how you might modify your workflow to include a conditional check:

```csharp
public class CustomWorkflow : IWorkflow
{
    public void Build(IWorkflowBuilder builder)
    {
        builder
            .StartWith<Activity1>()
            .Then<Activity2>()
            // Add a conditional activity here
            .If(context => context.GetInput<bool>("Flag"))
                .Then<Activity3>() // This will execute if the flag is true
            .Else()
                .Then<Activity4>() // This will execute if the flag is false
            .Then<Activity4>(); // Continue with Activity4 after Activity3 if the flag was true
    }
}
```

In this example, the `If` activity evaluates a lambda expression that retrieves the input named "Flag" from the workflow context. If the flag is `true`, `Activity3` will be executed; otherwise, the workflow will proceed directly to `Activity4`. After `Activity3` is executed (if the flag was true), the workflow will continue with `Activity4`.

Please note that the `context.GetInput<T>` method is used to retrieve the flag value from the workflow context. You will need to ensure that this flag is provided to the workflow when it is started or set as a variable within a previous activity.

If you need to perform more complex branching logic, you might consider using the `Switch` activity instead.

Remember to adjust the workflow continuation after the conditional branch to fit your specific requirements.

This comment was generated by [Glime](https://glimelab.ai/?utm_source=dRbvTb7IhCSYiI&utm_medium=github).

**Comment by NiveditaGalwade** [comment]
Getting below error 
.If(context => context.GetInput<bool>("Flag"))

Error	CS1929	'ISetupActivity<If>' does not contain a definition for 'GetInput' and the best extension method overload 'ExpressionExecutionContextExtensions.GetInput<bool>(ExpressionExecutionContext, InputDefinition)' requires a receiver of type 'Elsa.Expressions.Models.ExpressionExecutionContext'
